client/src/components/Verejne/Sidebar/Sidebar.js

```diff
@@ -0,0 +1,144 @@
+  autocompleteValueSelector,
+  autocompleteOptionsSelector,
+  openedAddressDetailSelector,
+  entitySearchValueSelector,
+  drawerOpenSelector,
+} from '../../../selectors'
+import {ENTITY_CLOSE_ZOOM, FIND_ENTITY_TITLE} from '../../../constants'
+import AddressDetail from './../Map/AddressDetail'
+import PlacesAutocomplete from '../../PlacesAutocomplete'
+
+const _DrawerIcon = ({drawerOpen, toggleDrawer}) =>
+  (drawerOpen ? <ArrowLeftIcon onClick={toggleDrawer} className="drawer-handle p-1"/> :
+    <ArrowRightIcon onClick={toggleDrawer} className="drawer-handle p-1" />)
+
+const DrawerIcon =
+  connect( (state) => ({
+      drawerOpen: drawerOpenSelector(state),
+  }),
+    {toggleDrawer}
+  )(_DrawerIcon)
+
+const _Content = ({
+  autocompleteValue,
+  setAutocompleteValue,
+  setZoomToLocation,
+  autocompleteOptions,
+  toggleModalOpen,
+  toggleDrawer,
+  openedAddressId,
+  entitySearchValue,
+  setEntitySearchValue,
+  findEntities,
+}) => (
+  <React.Fragment>
+    <Form onSubmit={findEntities}>
+      <FormGroup>
+        <Input
+          type="text"
+          placeholder={FIND_ENTITY_TITLE}
+          value={entitySearchValue}
+          onChange={setEntitySearchValue}
+        />
+      </FormGroup>
+    </Form>
+    <FormGroup>
+      <PlacesAutocomplete
+        value={autocompleteValue}
+        onSelect={setZoomToLocation}
+        onChange={setAutocompleteValue}
+        onError={(status, clearSuggestions) => clearSuggestions()}
+        searchOptions={autocompleteOptions}
+        className="form-control"
+      />
+    </FormGroup>
+    {openedAddressId && <AddressDetail addressId={openedAddressId} />}
+  </React.Fragment>
+)
+
+const Content = compose(
+  connect(
+    (state) => ({
+      autocompleteValue: autocompleteValueSelector(state),
+      autocompleteOptions: autocompleteOptionsSelector(state),
+      openedAddressId: openedAddressDetailSelector(state),
+      entitySearchValue: entitySearchValueSelector(state),
+    }),
+    {updateValue, zoomToLocation, toggleModalOpen, setEntitySearchFor, toggleDrawer}
+  ),
+  withHandlers({
+    findEntities: ({toggleModalOpen, setEntitySearchFor, entitySearchValue, toggleDrawer}) => (e) => {
+      e.preventDefault()
+      toggleModalOpen()
+      setEntitySearchFor(entitySearchValue)
+      toggleDrawer()
+    },
+    setAutocompleteValue: ({updateValue}) => (value) =>
+    updateValue(['publicly', 'autocompleteValue'], value, 'Set autocomplete value'),
+    setEntitySearchValue: ({updateValue}) => (e) =>
+    updateValue(['publicly', 'entitySearchValue'], (e.target.value), 'Set entity search field value'),
+    setZoomToLocation: ({zoomToLocation}) => (value, id) =>
+    geocodeByAddress(value)
+    .then((results) => getLatLng(results[0]))
+    .then((location) => zoomToLocation(location, ENTITY_CLOSE_ZOOM)),
+  })
+)(_Content)
+
+const Sidebar = ({
+  toggleDrawer,
+  closeDrawer,
+  drawerOpen,
+  renderDrawer,
+}) => (
+  renderDrawer ? <Drawer level={null}
```

Comment:
At first I tried to render both and then hide one with css, but i ran into some problems when interacting with Map.

Reply:
I thinks it performance-wise better this way.